desc/objectives/objective_funs.py

```diff
@@ -383,7 +384,13 @@ def unpack_state(self, x, per_objective=True):
         xs = jnp.split(x, xs_splits)
         params = [t.unpack_params(xi) for t, xi in zip(self.things, xs)]
         if per_objective:
+            # params is a list of lists of dicts, for each thing and for each objective
             params = self._unflatten(params)
+            # this filters out the params of things that are unused by each objective
+            params = [
```

Comment:
so after this is it still a list of lists? or just a list of dict?

Reply:
It's still a list of lists. The outermost list is for each objective, and the inner list is for each thing. Then within that it's a dict of the params for that thing. Like:
`[[{thing1_params}, {thing2_params}], [{thing1_params}, {thing2_params}]]`

At first, `params` contains all of the things for every objective. This logic is removing the things that aren't needed by that objective. So if the first objective only needs thing1 and the second objective only needs thing2, it will become:
`[[{thing1_params}], [{thing2_params}]]`